src/consensus/pbft/libbyz/LedgerWriter.h

```diff
@@ -4,22 +4,23 @@
 
 #include "Prepared_cert.h"
 #include "View_change.h"
-#include "consensus/ledgerenclave.h"
+#include "consensus/pbft/pbftpreprepares.h"
+#include "consensus/pbft/pbftrequests.h"
+#include "consensus/pbft/pbfttypes.h"
+#include "kv/kv.h"
 #include "ledger.h"
 #include "types.h"
 
 class LedgerWriter
 {
 private:
-  std::unique_ptr<consensus::LedgerEnclave> ledger;
+  pbft::Store& store;
+  pbft::PrePrepares& pbft_pre_prepares;
 
 public:
-  LedgerWriter(std::unique_ptr<consensus::LedgerEnclave> ledger_);
+  LedgerWriter(pbft::Store& store_, pbft::PrePrepares& pbft_pre_prepares_);
```

Comment:
I think it would be better to pass a reference to `PrePrepares` to the `pbft::Adaptor` in `nodestate.h` instead of passing it through here and `Replica`?

Reply:
`Replica` needs the tables anyway to set the hooks so it would make sense to pass the table here too. Right now the `pbft::Adaptor` doesn't store any state apart from the `Store` so I am not sure I would like to have the `Adaptor` keep a reference to a table.